src/components/Table/index.tsx

```diff
@@ -0,0 +1,260 @@
+import React, { useState } from 'react';
+import PerfectScrollbar from 'react-perfect-scrollbar';
+import {
+    Box,
+    Card,
+    Table,
+    TableBody,
+    TableCell,
+    TableHead,
+    TablePagination,
+    TableRow,
+    CardHeader,
+    Divider,
+    TextField,
+    InputAdornment,
+    SvgIcon,
+    TableSortLabel,
+} from '@material-ui/core';
+import orderByFunc from 'lodash/orderBy';
+import formatDate from '../../utils/DateFormat';
+import useStyles from './style';
+import { Search as SearchIcon } from 'react-feather';
+import { genPaginationLabel, applyPagination } from '../../utils/Pagenation';
+import applyFilter from '../../utils/FilterTable';
+
+interface TableColumnHeading<T> {
+    id: string;
+    width: string;
+    type?: 'date' | 'text' | 'number';
+    Cell?: React.FC<T>;
+    Head?: React.FC | string;
+    align: 'left' | 'center' | 'right' | 'justify' | 'inherit';
+}
+
+interface TableProps<T extends DefaultRow> {
+    columns: TableColumnHeading<T>[];
+    rows: T[];
+    title: string;
+    isPaginated: boolean;
+}
+interface DefaultRow {
+    id?: string;
+}
+
+export function POSTable<T extends DefaultRow>({
+    columns,
+    rows,
+    title,
+    isPaginated,
+}: TableProps<T>): JSX.Element {
+    const [page, setPage] = useState(0);
+    const [limit, setLimit] = useState(10);
+    const [query, setQuery] = useState('');
+    const [order, setOrder] = useState({
+        by: columns[0].id,
+        direction: true,
+    });
+
+    const classes = useStyles();
+
+    const handlePageChange = (event: unknown, newPage: number) => {
```

Comment:
you can change the unknown type and use better one

Reply:
okay will check it